stellargraph/core/element_data.py

```diff
@@ -279,8 +281,10 @@ def features(self, type_name, id_ilocs) -> np.ndarray:
             raise ValueError("unknown IDs")
 
         try:
-            return self._features[type_name][feature_ilocs, :]
-        except IndexError:
+            return tf.nn.embedding_lookup(
+                self._features[type_name], feature_ilocs.astype(int),
```

Comment:
Why does this now need casting?

Reply:
`node_ilocs` are of type `uint8`, unfortunately `embedding_lookup` only support `int32` and `int64`